vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <cmath>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/media/helper.h"
+#include "bat/ledger/internal/media/twitter.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace braveledger_media {
+
+MediaTwitter::MediaTwitter(bat_ledger::LedgerImpl* ledger):
+  ledger_(ledger) {
+}
+
+MediaTwitter::~MediaTwitter() {
+}
+
+// static
+std::string MediaTwitter::GetProfileURL(const std::string& screen_name) {
+  return base::StringPrintf("https://twitter.com/%s/", screen_name.c_str());
+}
+
+// static
+std::string MediaTwitter::GetProfileImageURL(const std::string& screen_name) {
+  return base::StringPrintf(
+      "https://twitter.com/%s/profile_image?size=original",
```

Comment:
is it possible to locally cache this image (like what chrome does for favicons) so that it's not fetched from twitter every time it needs to be loaded?

Reply:
I have some vague recollection that we did similar caching elsewhere @NejcZdovc?

Reply:
@diracdeltas we cache this image locally. So it's not used directly. This is entry in the DB

![image](https://user-images.githubusercontent.com/9574457/57565213-cef7c880-73b9-11e9-9a6a-f2126902ce84.png)